samples/tutorial/Tutorial3.md

```diff
@@ -0,0 +1,415 @@
+
+Because the parent workflow needs to know about the fact that `TodoListWorkflow` renders a list (so it can append its screens to the backstack), we will first need to modify the todo list workflow to return a _list_ of `Any`s as the rendering.
+
+```kotlin
+object TodoListWorkflow : StatefulWorkflow<ListProps, State, Back, List<Any>>() {
+
+  // …
+
+  override fun render(
+    props: ListProps,
+    state: State,
+    context: RenderContext
+  ): List<Any> {
+    val titles = state.todos.map { it.title }
+    val todoListScreen = TodoListScreen(
+        name = props.name,
+        todoTitles = titles,
+        onTodoSelected = { context.actionSink.send(selectTodo(it)) },
+        onBack = { context.actionSink.send(onBack()) }
+    )
+
+    return listOf(todoListScreen)
+  }
+}
+```
+
+Next, update the `RootWorkflow` to pass the name into the `TodoListWorkflow` and handle the `.back` output:
+
+```kotlin
+object RootWorkflow : StatefulWorkflow<Unit, State, Nothing, BackStackScreen<*>>() {
+
+  // …
+
+  override fun render(
+    props: Unit,
+    state: State,
+    context: RenderContext
+  ): BackStackScreen<*> {
+    // Our list of back stack items. Will always include the "WelcomeScreen".
+    val backstackScreens = mutableListOf<Any>()
+
+    // Render a child workflow of type WelcomeWorkflow. When renderChild is called, the
+    // infrastructure will create a child workflow with state if one is not already running.
+    val welcomeScreen = context.renderChild(WelcomeWorkflow) { output ->
+      // When WelcomeWorkflow emits LoggedIn, turn it into our login action.
+      login(output.name)
+    }
+    backstackScreens += welcomeScreen
+
+    when (state) {
+      // When the state is Welcome, defer to the WelcomeWorkflow.
+      is Welcome -> {
+        // We always add the welcome screen to the backstack, so this is a no op.
+      }
+
+      // When the state is Todo, defer to the TodoListWorkflow.
+      is Todo -> {
+        val todoListScreens = context.renderChild(TodoListWorkflow, ListProps(state.name)) {
+          logout()
+        }
+        backstackScreens.addAll(todoListScreens)
+      }
+    }
+
+    // Finally, return the BackStackScreen with a list of BackStackScreen.Items
+    return backstackScreens.toBackStackScreen()
+  }
+
+  // …
+}
+```
+
+Run the app again to validate it still behaves the same.
+
+### Adding the edit workflow as a child to the `TodoListWorkflow`
+
+Now that the `TodoListWorkflow`'s rendering is a list of screens, it can be updated to show the edit workflow when a `Todo` item is tapped.
+
+Modify the state to represent if the list is being viewed, or an item is being edited:
+
+```kotlin
+object TodoListWorkflow : StatefulWorkflow<ListProps, State, Back, List<Any>>() {
+
+  data class ListProps(val name: String)
+
+  data class State(
+    val todos: List<TodoModel>,
+    val step: Step
+  ) {
+    sealed class Step {
```

Comment:
Any good resources we can point to for forming the shape of State for common use cases? (I mean for even React or any uni-directional data flow state machine). It seems like this is one of the 'art' areas of using Workflow where extra resources will help.